sopel/modules/remind.py

```diff
@@ -226,47 +226,227 @@ def remind_in(bot, trigger):
+                month = int(date2)
+                day = int(date3)
+            else:
+                # dd-mm-YYYY or dd/mm/YY
+                year = int(date3)
+                month = int(date2)
+                day = int(date1)
+        elif date1 and date2:
+            if len(date1) == 4:
+                # YYYY-mm
+                year = int(date1)
+                month = int(date2)
+            elif len(date2) == 4:
+                # mm-YYYY
+                year = int(date2)
+                month = int(date1)
+            else:
+                # dd/mm
+                month = int(date2)
+                day = int(date1)
+
+        self.year = year
+        self.month = month
+        self.day = day
+
+    def __eq__(self, other):
+        return all(
+            getattr(self, attr) == getattr(other, attr, None)
+            for attr in [
+                'hour',
+                'minute',
+                'second',
+                'timezone',
+                'year',
+                'month',
+                'day',
+                'message',
+            ]
+        )
+
+    def get_duration(self, today=None):
+        """Get the duration between the reminder and ``today``
+
+        :param today: aware datetime to compare to, default to UTC now
+        :type today: aware :class:`datetime.datetime`
+        :return: The duration, in second, between ``today`` and the reminder.
+        :rtype: :class:`int`
+
+        This method returns the number of seconds given by the
+        :class:`datetime.timedelta` obtained by the difference between the
+        reminder and ``today``.
+
+        If the delta between the reminder and ``today`` is negative, Python
+        will represent it as a negative number of days, and a positive number
+        of seconds: since it returns the number of seconds, any past reminder
+        will be transformed into a future reminder the next day.
+        """
+        if not today:
+            today = datetime.now(self.timezone)
+        else:
+            today = today.astimezone(self.timezone)
+
+        year = self.year if self.year is not None else today.year
+        month = self.month if self.month is not None else today.month
+        day = self.day if self.day is not None else today.day
 
+        at_time = datetime(
+            year, month, day,
+            self.hour, self.minute, self.second,
+            tzinfo=today.tzinfo)
 
-REGEX_AT = re.compile(r'(\d+):(\d+)(?::(\d+))?([^\s\d]+)? (.*)')
+        timediff = at_time - today
+        duration = timediff.seconds
+
+        if timediff.days > 0:
+            duration = duration + (86400 * timediff.days)
+
+        return duration
+
+
+def parse_regex_match(match, default_timezone=None):
+    """Parse a time reminder from ``match``
+
+    :param match: :obj:`~.REGEX_AT`'s matching result
+    :param default_timezone: Default trigger's timezone
+                             (from the nick's, channel's, or config's timezone)
```

Comment:
I think the Franglish is so strong here that I can't even grok enough of what was meant to suggest an improvement. :/ In general, though (and this applies to the `match` param too), your parameter descriptions tend to lean really heavily on possessives, and that tends to make them confusing (or at least, makes them read strangely). Avoiding possessives as much as possible when writing these might help you create clearer descriptions. :)

Reply:
Hm... could you give me a description for the default timezone to use when `match` doesn't have one?

Reply:
I tried to go with that:

> timezone used when `match` doesn't have one; defaults to `UTC`